src/autoscaler/metricscollector/server/metric_handler.go

```diff
@@ -35,16 +35,42 @@ func NewMetricHandler(logger lager.Logger, cfc cf.CfClient, consumer noaa.NoaaCo
 func (h *MetricHandler) GetMetricHistories(w http.ResponseWriter, r *http.Request, vars map[string]string) {
 	appId := vars["appid"]
 	metricType := vars["metrictype"]
+	instanceIndexParam := r.URL.Query()["instanceindex"]
 	startParam := r.URL.Query()["start"]
 	endParam := r.URL.Query()["end"]
 	orderParam := r.URL.Query()["order"]
 
-	h.logger.Debug("get-metric-histories", lager.Data{"appId": appId, "metrictype": metricType, "start": startParam, "end": endParam, "order": orderParam})
+	h.logger.Debug("get-metric-histories", lager.Data{"appId": appId, "metrictype": metricType, "instanceIndex": instanceIndexParam, "start": startParam, "end": endParam, "order": orderParam})
 
 	var err error
 	start := int64(0)
 	end := int64(-1)
 	order := db.ASC
+	instanceIndex := int64(-1)
```

Comment:
same as above, any special reason to use int64 instead of int?

Reply:
@boyang9527 instanceIndex will be the result of strconv.ParseInt which is a int64, so define it int64 here and convert to int when calling db.RetrieveInstanceMetrics.